src/hotspot/share/opto/castnode.cpp

```diff
@@ -254,13 +251,6 @@ Node* CastIINode::Identity(PhaseGVN* phase) {
   if (progress != this) {
     return progress;
   }
-  if (_range_check_dependency) {
-    if (phase->C->post_loop_opts_phase()) {
-      return this->in(1);
```

Comment:
Does the removal of the CastII happen anywhere at all any more now?

Reply:
Yes,during final graph reshape.